client/src/components/MarketOrder.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+
+class MarketOrder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      power: '$1000.00',
+      view: 'buy',
+      shares: '',
+      estim: 0,
+      reviewOrder: 'null',
+      remaining: '',
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleEstimatedCost = this.handleEstimatedCost.bind(this);
+    this.renderReviewOrder = this.renderReviewOrder.bind(this);
+    this.handleReviewOrder = this.handleReviewOrder.bind(this);
+    this.backPress = this.backPress.bind(this);
+    this.handleBuy = this.handleBuy.bind(this);
+  }
+   
+  backPress(e) {
+    e.preventDefault();
+    this.setState({
+      reviewOrder: 'null'
+    })
+  }
+  
+  handleBuy(e) {
+    e.preventDefault();
+    let newPower = '$' + this.state.remaining;
+    this.setState({
+      power: newPower,
+      estim: 0,
+      shares: 0,
+      remaining: 0,
+      reviewOrder: 'null'
+    }, alert('You have successfully purchased your order!'))
+  }
+
+  handleChange(e) {
+    e.preventDefault();
+    this.setState({
+      [e.target.name]: Math.floor(Number(e.target.value))
+    }, this.handleEstimatedCost(Math.floor(Number(e.target.value))))
+  }
+  
+  handleEstimatedCost(share) {
+    let price = Number(this.props.stock.price.slice(1, this.props.stock.price.length));
+    let estimPrice = share * price;
+    let buyingPower = Number(this.state.power.slice(1, this.state.power.length))
+    let remaining = buyingPower - estimPrice
+    this.setState({
+      estim: estimPrice.toFixed(2),
+      remaining: remaining.toFixed(2)
+    })
+  }
+  
+  handleReviewOrder() {
+    let estim = Number(this.state.estim) * 1.05
+    let buyingPower = Number(this.state.power.slice(1, this.state.power.length))
+    if (estim > buyingPower) {
+      this.setState({
+        reviewOrder: 'false'
```

Comment:
Maybe I am missing something on these too, but just in case it wasn't intentional, a note that 'false' is truthy but false is falsy.